RealmSwift/Impl/SchemaDiscovery.swift

```diff
@@ -60,6 +60,28 @@ internal extension RLMProperty {
             self.updateAccessors()
         }
     }
+
+    /// Exposed for Macros.
+    /// Important: Keep args in same order & default value as `@Persisted` property wrapper
+    public convenience init<O: ObjectBase, V: _Persistable>(
+        name: String,
+        objectType _: O.Type,
+        valueType _: V.Type,
+        indexed: Bool = false,
+        primaryKey: Bool = false,
+        originProperty: String? = nil
+    ) {
+        self.init()
+        self.name = name
+        self.type = V._rlmType
+        self.optional = V._rlmOptional
+        self.indexed = primaryKey || indexed
+        self.isPrimary = primaryKey
+        self.linkOriginPropertyName = originProperty
+        V._rlmPopulateProperty(self)
+        V._rlmSetAccessor(self)
```

Comment:
This may be one of the things covered by the dyld cache on iOS, which means that it'll only be slow the first time an app is launched after each installation. If not, it is just frontloading work that'll need to happen at some point. Theoretically it could be done lazily, but right now we take advantage of that the schema is immutable after construction to share it between threads without any locking, and introducing locking could be a meaningful perf hit.